tests/integration/integration_python_test.go

```diff
@@ -341,3 +341,59 @@ func TestPythonPylint(t *testing.T) {
 	}
 	integration.ProgramTest(t, opts)
 }
+
+// Test remote component construction in Python.
+func TestConstructPython(t *testing.T) {
+	pathEnv, err := testComponentPathEnv()
+	if err != nil {
+		t.Fatalf("failed to build test component PATH: %v", err)
+	}
+
+	// TODO[pulumi/pulumi#5455]: Dynamic providers fail to load when used from multi-lang components.
+	// Until we've addressed this, set PULUMI_TEST_YARN_LINK_PULUMI, which tells the integration test
+	// module to run `yarn install && yarn link @pulumi/pulumi` in the Python program's directory, allowing
+	// the Node.js dynamic provider plugin to load.
+	// When the underlying issue has been fixed, the use of this environment variable inside the integration
+	// test module should be removed.
+	os.Setenv("PULUMI_TEST_YARN_LINK_PULUMI", "true")
+
+	var opts *integration.ProgramTestOptions
+	opts = &integration.ProgramTestOptions{
+		Env: []string{pathEnv},
+		Dir: filepath.Join("construct_component", "python"),
+		Dependencies: []string{
+			filepath.Join("..", "..", "sdk", "python", "env", "src"),
+		},
+		Quick: true,
+		ExtraRuntimeValidation: func(t *testing.T, stackInfo integration.RuntimeValidationStackInfo) {
+			assert.NotNil(t, stackInfo.Deployment)
+			if assert.Equal(t, 9, len(stackInfo.Deployment.Resources)) {
+				stackRes := stackInfo.Deployment.Resources[0]
+				assert.NotNil(t, stackRes)
+				assert.Equal(t, resource.RootStackType, stackRes.Type)
+				assert.Equal(t, "", string(stackRes.Parent))
+
+				// Check that dependencies flow correctly between the originating program and the remote component
+				// plugin.
+				urns := make(map[string]resource.URN)
+				for _, res := range stackInfo.Deployment.Resources[1:] {
+					assert.NotNil(t, res)
+
+					urns[string(res.URN.Name())] = res.URN
+					switch res.URN.Name() {
+					case "child-a":
+						for _, deps := range res.PropertyDependencies {
+							assert.Empty(t, deps)
+						}
+					case "child-b":
+						assert.Equal(t, []resource.URN{urns["a"]}, res.PropertyDependencies["echo"])
+					case "child-c":
+						assert.ElementsMatch(t, []resource.URN{urns["child-a"], urns["a"]},
+							res.PropertyDependencies["echo"])
```

Comment:
Note: These cases are slightly different than the Node test, because our Node.js SDK filters out ComponentResources from dependencies, but Python does not.

https://github.com/pulumi/pulumi/blob/2585b86aa4252b1ea430dbfd2324c47ea57763b1/sdk/nodejs/runtime/resource.ts#L392-L419

So for Python, we need to include the component resources here.